sdk/trace/provider.go

```diff
@@ -327,7 +328,10 @@ func WithIDGenerator(g IDGenerator) TracerProviderOption {
 // Tracers the TracerProvider creates to make their sampling decisions for the
 // Spans they create.
 //
-// If this option is not used, the TracerProvider will use a
+// This option overrides the Sampler configured through the OTEL_TRACES_SAMPLER
+// and OTEL_TRACES_SAMPLER_ARG environment variables. If this option is not used
```

Comment:
What do you think about also documenting these env vars also in the `NewTracerProvider` GoDoc comment?

Reply:
Alternatively, you can make the env vars so that they exported like here: https://github.com/open-telemetry/opentelemetry-go/pull/2515#discussion_r787921391

Reply:
Since `opentelemetry-go` still doesn't support all TracerProvider environment variables, I decided to only document the relevant bits touched by this PR.